core/block/editor/smartblock/smartblock.go

```diff
@@ -890,27 +890,74 @@ func (sb *smartBlock) injectLinksDetails(s *state.State) {
 }
 
 func (sb *smartBlock) injectLocalDetails(s *state.State) error {
-	if pbtypes.GetString(s.LocalDetails(), bundle.RelationKeyWorkspaceId.String()) == "" {
-		wsId, err := sb.coreService.GetWorkspaceIdForObject(sb.Id())
-		if wsId != "" {
-			s.SetDetailAndBundledRelation(bundle.RelationKeyWorkspaceId, pbtypes.String(wsId))
-		} else if pbtypes.GetBool(s.LocalDetails(), bundle.RelationKeyIsDeleted.String()) && err != core.ErrObjectDoesNotBelongToWorkspace {
-			log.With("objectID", sb.Id()).Warnf("injectLocalDetails empty workspace: %v", err)
-		}
+	sb.injectWorkspaceID(s)
+
+	details, err := sb.getDetailsFromStore()
+	if err != nil || details == nil {
+		return err
 	}
 
-	if sb.objectStore == nil {
+	sb.injectBackLinks(details)
+	hasPendingLocalDetails := sb.updatePendingDetails(details)
+
+	// inject also derived keys, because it may be a good idea to have created date and creator cached,
+	// so we don't need to traverse changes every time
+	keys := append(bundle.LocalRelationsKeys, bundle.DerivedRelationsKeys...)
+	storedLocalScopeDetails := pbtypes.StructFilterKeys(details, keys)
+	sbLocalScopeDetails := pbtypes.StructFilterKeys(s.LocalDetails(), keys)
+	if pbtypes.StructEqualIgnore(sbLocalScopeDetails, storedLocalScopeDetails, nil) {
 		return nil
 	}
+
+	s.InjectLocalDetails(storedLocalScopeDetails)
+	if p := s.ParentState(); p != nil && !hasPendingLocalDetails {
+		// inject for both current and parent state
+		p.InjectLocalDetails(storedLocalScopeDetails)
+	}
+
+	return sb.injectCreationInfo(s)
+}
+
+func (sb *smartBlock) getDetailsFromStore() (*types.Struct, error) {
+	if sb.objectStore == nil {
+		return nil, nil
+	}
 	storedDetails, err := sb.objectStore.GetDetails(sb.Id())
+	if err != nil || storedDetails == nil {
+		return nil, err
+	}
+	return storedDetails.GetDetails(), nil
+}
+
+func (sb *smartBlock) injectWorkspaceID(s *state.State) {
+	if pbtypes.GetString(s.LocalDetails(), bundle.RelationKeyWorkspaceId.String()) != "" {
+		return
+	}
+	wsID, err := sb.coreService.GetWorkspaceIdForObject(sb.Id())
+	if wsID != "" {
+		s.SetDetailAndBundledRelation(bundle.RelationKeyWorkspaceId, pbtypes.String(wsID))
+	} else if pbtypes.GetBool(s.LocalDetails(), bundle.RelationKeyIsDeleted.String()) && err != core.ErrObjectDoesNotBelongToWorkspace {
+		log.With("objectID", sb.Id()).Warnf("injectLocalDetails empty workspace: %v", err)
+	}
+}
+
+func (sb *smartBlock) injectBackLinks(details *types.Struct) {
+	if pbtypes.GetStringList(details, bundle.RelationKeyBacklinks.String()) != nil {
```

Comment:
Shouldn't we update this relation in case of new links?

Reply:
Yeah, of course we should!
Thank you for noticing this severe mistake!